packages/commercetools/api-client/src/api/getProduct/index.ts

```diff
@@ -26,14 +26,19 @@ const getProduct = async (context, params, customQuery?: CustomQuery) => {
     customQuery, { products: { query: defaultQuery, variables: defaultVariables } }
   );
 
-  const request = await (context.client as ApolloClient<any>).query<ProductData>({
-    query: gql`${products.query}`,
-    variables: products.variables,
-    // temporary, seems like bug in apollo:
-    // @link: https://github.com/apollographql/apollo-client/issues/3234
-    fetchPolicy: 'no-cache'
-  });
-  return request;
+  try {
+    const request = await (context.client as ApolloClient<any>).query<ProductData>({
+      query: gql`${products.query}`,
+      variables: products.variables,
+      // temporary, seems like bug in apollo:
+      // @link: https://github.com/apollographql/apollo-client/issues/3234
+      fetchPolicy: 'no-cache'
+    });
+    return request;
+  } catch (error) {
+    throw error.graphQLErrors?.[0] || error.networkError?.result || error;
```

Comment:
What does this try...catch it give us?

Reply:
Whole try catch allows us to catch an error that is thrown by GraphQL in the terminal but is not handled on the client side. This strange looking throw catches different parts of the error for different problems. The problem with GrapQL query limit is thrown in the NetworkError object of error, while bug with product id lives in GraphQLErrors. Normally it would only throw an error property or the whole error object, but here, with one change two problems can be fixed as they relate to each other.